src/Particles/babylon.particleSystem.ts

```diff
@@ -388,20 +499,57 @@
                 }
             }
 
+            // Animation sheet
+            var deltaTime = 0;
+            if (this.cellSize) {
+                var baseSize = this.particleTexture.getBaseSize();
+                var rowSize = baseSize.width / this.cellWidth;
+                var engine = this._scene.getEngine();
+                deltaTime = engine.getDeltaTime();
+
+                if (this._animationStarted) {
+                    this.appendParticleVertexes = this.appenedAllParticleVertexesWithSheetAndAnimationStarted;
+                }
+                else {
+                    this.appendParticleVertexes = this.appenedAllParticleVertexesWithSheet;
+                }
+            }
+            else {
+                this.appendParticleVertexes = this.appenedParticleVertexesNoSheet;
+            }
+
             // Update VBO
             var offset = 0;
             for (var index = 0; index < this.particles.length; index++) {
                 var particle = this.particles[index];
-
-                this._appendParticleVertex(offset++, particle, 0, 0);
-                this._appendParticleVertex(offset++, particle, 1, 0);
-                this._appendParticleVertex(offset++, particle, 1, 1);
-                this._appendParticleVertex(offset++, particle, 0, 1);
+                this.appendParticleVertexes(offset, particle, rowSize, deltaTime);

```

Comment:
Logic here could be a bit not straight forward but I learned from previous project to minimize conditions inside loops :)